### PR TITLE
xds, wrr: randomize the initial deadline in the scheduler

### PR DESCRIPTION
**Pull request by YifeiZhuang**
follow up change #9873

cc. @larry-safran

### CODE REVIEW

File: xds/src/test/java/io/grpc/xds/WeightedRoundRobinLoadBalancerTest.java
Comment (larry-safran) [inline xds/src/test/java/io/grpc/xds/WeightedRoundRobinLoadBalancerTest.java:678]: You need to set the seed in a constructor rather than each time you get a new value.  Otherwise you'll keep getting the same value.

File: xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java
Comment (ejona86) [inline xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java:382]: Is there a reason for `1-randomDouble` instead of just `randomDouble`? I see that flips the range to be inclusive of 1 and exclusive of 0, but that seems to not matter here. Shouldn't matter either way, but seemed strange.

**Review by ejona86 — COMMENTED** [review]
Larry's comment is important. But other than that, LGTM. I'll let him approve when that fix is made.

File: xds/src/test/java/io/grpc/xds/WeightedRoundRobinLoadBalancerTest.java
Comment (YifeiZhuang) [reply xds/src/test/java/io/grpc/xds/WeightedRoundRobinLoadBalancerTest.java:678]: I do want to make it get the same value - this basically disables the initial randomization. I did that because some tests are unhappy.

There are tests that enables it - it works because those tests runs 1000 times.

Comment (ejona86) [reply xds/src/test/java/io/grpc/xds/WeightedRoundRobinLoadBalancerTest.java:678]: Then just return a constant. No need to call `setSeed()`.

Comment (larry-safran) [reply xds/src/test/java/io/grpc/xds/WeightedRoundRobinLoadBalancerTest.java]: If some tests are unhappy, you should fix the tests.  What you have is the same as
  public double nextDouble() {
    return 0.7308781907032909;
  }
It doesn't matter how many times it is run and isn't really testing the randomness you want.  

I think what you really want is to pass it a list of numbers to return and when it gets to the end it just starts over.  That would give you the flexibility to really test that different weights and different orderings are being handled correctly.

Comment (YifeiZhuang) [reply xds/src/test/java/io/grpc/xds/WeightedRoundRobinLoadBalancerTest.java]: Determination is necessary to evaluate certain behavior, e.g. earlier deadline is picked first, same weight items order breaks the tie.  So I intentionally inject fake random there.
Different weights with randomization is tested in, e.g. `edfScheduler` test case.

**Review by larry-safran — APPROVED** [review]

**Pull request merged**
